src/Paprika/Data/NibblePath.cs

```diff
@@ -517,7 +517,10 @@ public bool Equals(in NibblePath other)
 
         return FindFirstDifferentNibble(other) == Length;
     }
-
+    public bool StartsWith(in NibblePath other)
+    {
+        return FindFirstDifferentNibble(other) > 0;
```

Comment:
If two path of length `3` are different at `2`, they do not start with each other, do they? 👀

Reply:
I have changed the implementation to  - 

return FindFirstDifferentNibble(other) == Length 

Does this look good?